spanner/value.go

```diff
@@ -71,44 +71,44 @@ var (
 // Encoder is the interface implemented by a custom type that can be encoded to
 // a supported type by Spanner. A code example:
 //
-// type customField struct {
-//     Prefix string
-//     Suffix string
-// }
+//   type customField struct {
```

Comment:
nit: It would be nice to turn these into a first class godoc example. Here is an example from the storage library:
- Godoc: https://pkg.go.dev/cloud.google.com/go/storage#example-ACLHandle.Delete
- Code: https://github.com/googleapis/google-cloud-go/blob/a5bbdb584c82f628069a5c0975e273ea45273782/storage/example_test.go#L527-L537

Reply:
I am not sure if this is a good case for that. There are two reasons to put in this way: 

* The first class godoc example seems to be a main function to guide users how to use the api. 
* The example can be close to the code. Probably better in the comment instead of jumping to another file.

Reply:
Thanks for suggestion! This example contains a type definition and method definition, so if I understand correctly, we need a separate file to turn this into first class godoc example.

For example, sort package has a similar example for [Example (SortWrapper)](https://golang.org/pkg/sort/#pkg-overview) by [this code](https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/sort/example_wrapper_test.go). I'm wondering if I should do similar thing for this example.